core/src/main/java/com/linecorp/armeria/common/TlsProvider.java

```diff
@@ -39,6 +47,59 @@ static TlsProvider of(TlsKeyPair tlsKeyPair) {
         return builder().keyPair(tlsKeyPair).build();
     }
 
+    /**
+     * Returns a {@link TlsProvider} that periodically refreshes the {@link TlsKeyPair} based on the given
+     * {@code interval}.
+     *
+     * <p>Note that the key pair supplier is also invoked during the initialization of the returned
+     * {@link TlsProvider}, so this factory method should not be called from an event loop thread.
+     */
+    static TlsProvider ofScheduled(Supplier<TlsKeyPair> keyPairSupplier, Duration interval) {
+        return ofScheduled(keyPairSupplier, interval, CommonPools.blockingTaskExecutor());
+    }
+
+    /**
+     * Returns a {@link TlsProvider} that periodically refreshes the {@link TlsKeyPair} based on the given
+     * {@code interval} using the specified {@link ScheduledExecutorService}.
+     *
+     * <p>Note that the key pair supplier is also invoked during the initialization of the returned
+     * {@link TlsProvider}, so this factory method should not be called from an event loop thread.
+     */
+    static TlsProvider ofScheduled(Supplier<TlsKeyPair> keyPairSupplier, Duration interval,
+                                   ScheduledExecutorService executor) {
+        return ofScheduled(keyPairSupplier, ImmutableList.of(), null, interval, executor);
+    }
+
+    /**
+     * Returns a {@link TlsProvider} that periodically refreshes the {@link TlsKeyPair} based on the given
+     * {@code interval} using the specified {@link ScheduledExecutorService}.
+     *
+     * <p>Note that the key pair supplier is also invoked during the initialization of the returned
+     * {@link TlsProvider}, so this factory method should not be called from an event loop thread.
+     *
+     * @param keyPairSupplier the supplier of {@link TlsKeyPair} to be used for TLS handshakes.
+     * @param certificates the trusted {@link X509Certificate}s to be used for verifying the remote endpoint's
+     *                     certificate.
+     * @param onKeyPairUpdated a {@link Consumer} that is invoked when the {@link TlsKeyPair} is updated.
+     * @param interval the interval at which the {@code keyPairSupplier} is invoked to refresh the
+     *                 {@link TlsKeyPair}.
+     * @param executor the {@link ScheduledExecutorService} that is used to schedule the periodic refresh.
+     */
+    static TlsProvider ofScheduled(Supplier<TlsKeyPair> keyPairSupplier,
+                                   Iterable<? extends X509Certificate> certificates,
```

Comment:
Note) I think the current implementation is fine, but once xDS mTLS implementation starts it is probably likely that both `TlsKeyPair` and `certificates` are both dynamically loaded